File: frontend/lib/models/rating.dart
class Rating {
  final String id;
  final String inquiryId;
  final int rating;
  final String? review;

  Rating({
    required this.id,
    required this.inquiryId,
    required this.rating,
    this.review,
  });

  factory Rating.fromJson(Map<String, dynamic> json) {
    return Rating(
      id: json['_id'] ?? '',
      inquiryId: json['inquiry_id'] is Map
          ? json['inquiry_id']['_id']
          : json['inquiry_id'] ?? '',
      rating: json['rating'] ?? 0,
      review: json['review'],
    );
  }
}

File: backend/models/Rating.js
const mongoose = require('mongoose');

const ratingSchema = new mongoose.Schema({
  inquiry_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Inquiry', required: true, unique: true },
  customer_id: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  rating: { type: Number, required: true, min: 1, max: 5 },
  review: { type: String, default: '' },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Rating', ratingSchema);
